docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +89,1015 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+         "silverstripe/admin": "~1.1.0",
+         "silverstripe/asset-admin": "~1.1.0",
+         "silverstripe/campaign-admin": "~1.1.0",
+         "silverstripe/errorpage": "~1.1.0",
+         "silverstripe/graphql": "~1.1.0",
+         "silverstripe/versioned": "~1.1.0"
+    }
+}
+```
 
-For more details, please read through the [`index.php` and `.htaccess` rewrites section](/changelogs/4.0.0#index-php-rewrites)
+#### Updating third party dependencies
+If you project requires third party modules, you'll need to adjust their associated constraint. This will allow you to retrieve a SilverStripe 4 compatible version of the module.
 
-After installing, run the upgrader doctor command:
+[Look up the package  on Packagist](https://packagist.org/) to see if a SilverStripe 4 version is provided.
 
+Take for example the following SilverStripe 3 `composer.json`.
+```json
+{
+    "name": "acme/cms-website",
+    "require": {
+        "silverstripe/framework": "3.6.5@stable",
+        "silverstripe/cms": "3.6.5@stable",
+        "dnadesign/silverstripe-elemental": "^1.8.0"
+    }
+}
 ```
-cd ~/my-project-root
-upgrade-code doctor
+
+Looking at the [Packagist entry for `dnadesign/silverstripe-elemental`](https://packagist.org/packages/dnadesign/silverstripe-elemental#2.0.0), you can see that versions 2.0.0 and above of this package are compatible with SilverStripe 4. So you can update that constraint to `^2.0.0`.
+
+Alternatively, you can set a very permissive constraint and let composer find a SilverStripe 4 compatible version. After you're done updating your dependencies, make sure you adjust your constraints to be more specific.
+
+Once you've updated your third-party packages constraints, try updating your dependencies by running `composer update`. If composer can't find all your dependencies it will throw an error.
+
+### Resolving conflicts
+Whatever you have updated your dependencies with the upgrader or manually, you'll likely have some conflicts to resolve. Running a `composer update` will tell you which package are conflicted and suggested alternative combinations of package that might work.
+
+The most typical reason for a conflict is that the maintainer of a package has not release a version compatible with SilverStripe 4.
+
+If the maintainer of the package is in the process of upgrading to SilverStripe 4, a development version of the package might be available. In some cases, it can be worthwhile to look up the repository of the package or to reach out to the maintainer.
+
+To resolve a conflict you can either:
+* remove the package from your project, if it is not essential
+* fork the package and maintain it yourself.
+
+If you decide to fork a package, remove it from your `composer.json` file and from your `.gitignore` file. Then track the package's codebase in your project source control. Alternatively, you can track the fork as a private  package if you do not which to merge it in your code.
```

Comment:
I'll need to think about that one a bit. I think a lot of modules that are not compatible with SS4 are probably abandoned. So doing a PR might not be an option in many cases.

Plus, I don't really want to turn this into a "create your first SS4 module" guide ... while it might be interesting to create that guide later on.

Reply:
OK - then I just think the language needs to change; rather than using the word "forking" (which to me implies public maintaining) I'd just say to copy the code into their own codebase and work on it.

Still, I'd put this below recommending raising a PR to upgrade a 3rd party package

Reply:
Good suggestion. I've differentiated between forking a module and integrating a module into your own codebase.

I've linked to our _How to publish a module_ doc page.